Sources/SentrySwiftUI/SentryTracedView.swift

```diff
@@ -39,61 +37,112 @@ import SwiftUI
+    public init(_ viewName: String? = nil, @ViewBuilder content: @escaping () -> Content) {
+        self.content = content
+        self.name = viewName ?? SentryTracedView.extractName(content: Content.self)
+        self.nameSource = viewName == nil ? .component : .custom
+    }
+#endif
+    
     private static func extractName(content: Any) -> String {
         var result = String(describing: content)
-
+        
         if let index = result.firstIndex(of: "<") {
             result = String(result[result.startIndex ..< index])
         }
-
+        
         return result
     }
-
+    
     public var body: some View {
-        if viewAppeared {
-            return self.content().onAppear()
+        var trace: SentryTracer?
+        var spanId: SpanId?
+        
+        if !viewAppeared {
+            trace = ensureTransactionExists()
+            spanId = createAndPushBodySpan(transactionCreated: trace != nil)
         }
-
-        var transactionCreated = false
-        if SentryPerformanceTracker.shared.activeSpanId() == nil {
-            transactionCreated = true
-            let transactionId = SentryPerformanceTracker.shared.startSpan(withName: self.name, nameSource: nameSource, operation: "ui.load", origin: self.traceOrigin)
-            SentryPerformanceTracker.shared.pushActiveSpan(transactionId)
-
-            //According to Apple's documentation, the call to `body` needs to be fast
-            //and can be made many times in one frame. Therefore they don't use async code to process the view.
-            //Scheduling to finish the transaction at the end of the main loop seems the least hack solution right now.
-            //'onAppear' is not a suitable place to do this because it may happen before other view `body` property get called.
-            DispatchQueue.main.async {
-                SentryPerformanceTracker.shared.popActiveSpan()
-                SentryPerformanceTracker.shared.finishSpan(transactionId)
-            }
-        }
-
-        let id = SentryPerformanceTracker.shared.startSpan(withName: transactionCreated ? "\(self.name) - body" : self.name, nameSource: nameSource, operation: "ui.load", origin: self.traceOrigin)
-
-        SentryPerformanceTracker.shared.pushActiveSpan(id)
+        
         defer {
-            SentryPerformanceTracker.shared.popActiveSpan()
-            SentryPerformanceTracker.shared.finishSpan(id)
+            if let spanId = spanId {
+                finishSpan(spanId)
+            }
         }
+        
+        return content()
+            .onAppear { viewAppeared = true }
+#if canImport(SwiftUI) && canImport(UIKit) && os(iOS) || os(tvOS)
+            // We need to add a UIView to the view hierarchy to be able to
+            // monitor ui life cycles. We will use the background modifier
+            // to add this tracking view behind the content.
+            .background(TracingView(name: self.name, waitForFullDisplay: self.waitforFullDisplay, tracer: trace))
```

Comment:
`h`: Maybe that's a dump question 😅: I don't understand why we can't use `.onAppear` and we have to use a UIView. Why do we need to monitor UI life cycles? The `SentryTimeToDisplayTracker` waits for the next frame to be drawn anyway. Please elaborate, or we can discuss this on a call.

Reply:
It's not a dump question at all. 
And actually I had something else in mind when I started the PR that was not possible. .onAppear may be good enough for this purpose.

Reply:
Excellent. This would be great because the code won't depend on UIKit directly.